portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,48 +14,112 @@
+        // Send the JSON as the response
+        response.setContentType("application/json;");
+        response.getWriter().println(convertToJson(comments.getHistory()));
+        
+    }
 
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    ArrayList<String> messages = new ArrayList<>();
-    messages.add("Jeff Dean writes directly in binary. He then writes the source code as documentation for other developers.");
-    messages.add("Unsatisfied with constant time, Jeff Dean created the world's first O(1/N) algorithm.");
-    messages.add("Jeff Dean's keyboard has two keys: 1 and 0.");
-
-    // Convert the ArrayList to JSON
-    String json = convertToJson(messages);
-
-    // Send the JSON as the response
-    response.setContentType("application/json;");
-    response.getWriter().println(json);
-  }
-
-
-  /**
-   * Converts a List of Strings into a JSON string using manual String concatentation.
-   */
-  private String convertToJson(ArrayList<String> messages) {
-    String json = "{";
-    json += "\"firstMessage\": ";
-    json += "\"" + messages.get(0) + "\"";
-    json += ", ";
-    json += "\"secondMessage\": ";
-    json += "\"" + messages.get(1) + "\"";
-    json += ", ";
-    json += "\"thirdMessage\": ";
-    json += "\"" + messages.get(2) + "\"";
-    json += "}";
-    return json;
-  }
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+        String comment = request.getParameter("comment");
+        String name = request.getParameter("name");
+        System.out.println("Trying to int parse " + request.getParameter("numComments"));
+
+        // Convert the input to an int.
+        int numComments;
+        try {
+            numComments = Integer.parseInt(request.getParameter("numComments"));
+        } catch (NumberFormatException e) {
+            System.err.println("Could not convert to int: " + request.getParameter("numComments"));
+        }
+
+        long timestamp = System.currentTimeMillis();
+
+        // Treat empty name as default name
+        if (name.equals("")) {
+            name = defaultName;
+        }
+
+        Entity taskEntity = new Entity("Task");
+        taskEntity.setProperty("name", name);
+        taskEntity.setProperty("comment", comment);
+        taskEntity.setProperty("timestamp", timestamp);
+
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        datastore.put(taskEntity);
+
+        // Log the comment and redirect back to the HTML page.
+        response.sendRedirect("/index.html");
+    }
+
+
+    /**
+    * Converts a List of Strings into a JSON string using the Gson library. Note: We first added
+    * the Gson library dependency to pom.xml.
+    */
+    private String convertToJson(List<String> messages) {
+        Gson gson = new Gson();
+        String json = gson.toJson(messages);
```

Comment:
feel like new Gson().toJson(msgs); is good enough?
And were we to introduce an util function, make it static and in a separate class

Reply:
My justification for this is that this method is copy-and-pasted from the example code that we are provided in the tutorial. However, if need be, I can change it.